### PR TITLE
[KOGITO-4457] Removing knative-eventing-addon

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-4457

In this PR we completely remove `knative-eventing-addon`, a module added in the Dark Ages to give support to CloudEvents over HTTP.

Now, the Quarkus CloudEvents Addon automatically adds the possibility to consume CE in the application's root path. It also plays nicely with any Smallrye Connector. Meaning that the service can consume from any Smallrye implementation/connector and also from HTTP CE.

Requires https://github.com/kiegroup/kogito-runtimes/pull/1124
Once CI gets back, we merge #1124, I rebase and we will have a more stable code base regarding CE handling.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
@fjtirado FYI

**Comment by ricardozanini** [comment]
Examples updated: https://github.com/kiegroup/kogito-examples/pull/643

**Review by cristianonicolai — APPROVED** [review]

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java [hunk 1]
```diff
@@ -15,35 +15,79 @@
...
     @Produces
     @ApplicationScoped
     @Named(KogitoEventStreams.PUBLISHER)
-    public Multi<String> makeMulti() {
-        return events
-                .invoke(Message::ack)
-                .map(Message::getPayload)
-                .broadcast().toAllSubscribers();
+    public Multi<String> producerFactory() {
+        return processor.onItem().transform((item) -> item);
```
Comment (evacchi) [inline addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java:59]: you can just return `processor`

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java [hunk 2]
```diff
@@ -15,35 +15,79 @@
...
+        LOGGER.debug("Received message from channel {}: {}", KogitoEventStreams.INCOMING, message);
+        return message
+                .ack()
+                .exceptionally((e) -> {
+                    LOGGER.error("Failed to ack message", e);
+                    return null;
+                })
+                .thenApply((r) -> {
+                    LOGGER.debug("Producing message from channel to internal bus: {}", message);
+                    processor.onNext(message.getPayload());
```
Comment (evacchi) [inline addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java:79]: maybe you wanted to use this?

```suggestion
                    produce(message.getPayload());
```

**Review by evacchi — APPROVED** [review]
couple of minor changes

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java (hunk 1 above)
Comment (fjtirado) [inline addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java:59]: You can remove parentheses around item and as Edoardo is requesting just return processor

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java [hunk 3]
```diff
@@ -15,35 +15,79 @@
...
+     *
+     * @param message the given message in JSON format
+     * @return a {@link CompletionStage} after ack-ing the message
+     */
+    @Incoming(KogitoEventStreams.INCOMING)
+    public CompletionStage<Void> onEvent(Message<String> message) {
+        LOGGER.debug("Received message from channel {}: {}", KogitoEventStreams.INCOMING, message);
+        return message
+                .ack()
+                .exceptionally((e) -> {
```
Comment (fjtirado) [inline addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java:73]: you can remove parentheses around e

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java [hunk 4]
```diff
@@ -15,35 +15,79 @@
...
+                    LOGGER.error("Failed to ack message", e);
+                    return null;
+                })
+                .thenApply((r) -> {
```
Comment (fjtirado) [inline addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusCloudEventPublisher.java:77]: same here, unneeded parentheses

**Review by fjtirado — COMMENTED** [review]
@ricardozanini thanks for sharing.
Just minor comestic issues
Although not part of this JIRA, since we are chaning kogitoProcessExtension, can we replace current getExtension implementation by
```
    private String getExtension(CloudEventExtensions extensions, String key) {
        return Objects.toString(extensions.getExtension(key));
    };
```
?

**Comment by ricardozanini** [comment]
> @ricardozanini thanks for sharing.
> Just minor comestic issues
> Although not part of this JIRA, since we are chaning kogitoProcessExtension, can we replace current getExtension implementation by
> 
> ```
>     private String getExtension(CloudEventExtensions extensions, String key) {
>         return Objects.toString(extensions.getExtension(key));
>     };
> ```
> 
> ?

I think it's better to leave as it's. If `null` we are returning an empty String instead of `null` String.

**Comment by fjtirado** [comment]
> > @ricardozanini thanks for sharing.
> > Just minor comestic issues
> > Although not part of this JIRA, since we are chaning kogitoProcessExtension, can we replace current getExtension implementation by
> > ```
> >     private String getExtension(CloudEventExtensions extensions, String key) {
> >         return Objects.toString(extensions.getExtension(key));
> >     };
> > ```
> > 
> > 
> > ?
> 
> I think it's better to leave as it's. If `null` we are returning an empty String instead of `null` String.

Ok, you can use
  private String getExtension(CloudEventExtensions extensions, String key) {
        return Objects.toString(extensions.getExtension(key),"");
    };

**Comment by ricardozanini** [comment]
LOL

**Comment by ricardozanini** [comment]
Jenkins retest this

**Review by danielezonca — APPROVED** [review]

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/events/CloudEventListenerResourceQuarkusTemplate.java [hunk 5]
```diff
@@ -60,7 +59,8 @@ void setup() {
         try {
             LOGGER.debug("CloudEvent received: {}", Printer.beautify(event));
             // convert CloudEvent to JSON and send to internal channels
-            emitter.send(objectMapper.writeValueAsString(event));
+            //emitter.send(objectMapper.writeValueAsString(event));
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/events/CloudEventListenerResourceQuarkusTemplate.java:62]: Yes, I'll rebase after #1124

File: addons/cloudevents/cloudevents-quarkus-addon-it/src/test/java/org/kie/kogito/integrationtests/CloudEventListenerResourceTest.java [hunk 6]
```diff
@@ -29,8 +29,8 @@
 
 import io.cloudevents.jackson.JsonFormat;
 import io.quarkus.test.junit.QuarkusTest;
-import io.restassured.RestAssured;
 
+import static io.restassured.RestAssured.*;
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply addons/cloudevents/cloudevents-quarkus-addon-it/src/test/java/org/kie/kogito/integrationtests/CloudEventListenerResourceTest.java:33]: IDE thing. Sure, I'll remove it. Haven't noticed.

**Review by fjtirado — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Comment by danielezonca** [comment]
@ricardozanini
Minor thing, can you please rename the PR? `Deprecating` is not correct considering that we are removing it completely

**Comment by ricardozanini** [comment]
Jenkins retest this

**Comment by cristianonicolai** [comment]
@ricardozanini needs rebase

**Pull request merged**